impeller/aiks/canvas.cc

```diff
@@ -407,13 +411,14 @@ bool Canvas::AttemptDrawBlurredRRect(const Rect& rect,
     // Defer the alpha, blend mode, and image filter to a separate layer.
     SaveLayer({.color = Color::White().WithAlpha(rrect_color.alpha),
                .blend_mode = paint.blend_mode,
-               .image_filter = paint.image_filter});
+               .image_filter = paint.image_filter},
+              rect, nullptr, ContentBoundsPromise::kContainsContents, 1u);
```

Comment:
Actually since this is for the rrect blur we need to expand the coverage size up to some threshold. Doesn't need to be done now I'd just note it so we don't forget.

Reply:
err not rrect blur, the slow gaussian blur path.

Reply:
The bounds for a SaveLayer are the content bounds, not for the filtered bounds, so it should not include any expansion by the image_filter in the SaveLayer paint...

Reply:
Or, are you saying that the contents of this SaveLayer will be the expanded blurred rrect?

Reply:
Ahh right, if its the content bounds its just the rect.

Reply:
No, I think it is more sometimes. The thing we render into this SaveLayer is the blurred rect. The ImageFilter gets applied to the post-blurred rect. I need to write some test cases to verify. Also, I want to write some cases that ensure our depth allocation for these save[Layer] calls don't cause us to exceed depth budgets somewhere else.

Reply:
It would probably better if we did this sort of lowering during recording instead of during replay, then from the dispatcher perspective we don't need to worry about computing any bounds/depth values.

Reply:
Lowering? I'm not sure what that term means in this context.

There is a lot in this method that is completely dependent on how we've implemented the blurs today that might not be true in the near future. In particular, I have on my plate a blur algorithm that can easily do all BlurStyles with a single shader, but I need to update it to deal with non-square corners...

Reply:
loweing as in replacing things like `drawShadow` with `saveLayer/blur drawShape restore`. It would be specific to this renderer but that is good IMO

Reply:
Also, as we progress along with the experimental canvas we might start relying less on calling one method from another with depth implications and just insert the associated rendering structures more directly. Then we'll have more immediate control over the allocation of depths from each operation. Currently, the things that you suggest will have automatic allocations of depth implicit in calling "that other rendering operation". We could create a new layer, draw the blurred thing to it, and then render it back to the original render target all without needing to bump the depth more than the implicitly assigned single depth value. For now there are just a lot of side effects to each implementation because we're trying to leverage the regular canvas ops.

Reply:
I'm going to adjust for this by just having the DL allocate 1 depths per rendering op when there is a MaskFilter in effect. It's kind of Impeller-specific logic, but it's abstracted a bit to keep the object encapsulation intact. In reality, DL could always allocate 5 or 20 depths to each rendering op and we'd still never really run out of room in the depth buffer. Then we won't care how efficient or multi-staged the Impeller implementations are. For now, I'm trying to keep it mostly stingy with occasional conservative over-allocations...

Reply:
Seems reasonable 👍

Reply:
The latest commit adjusts for this additional SaveLayer.